File: Data Structures & Algorithms/two-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        
        let map = {}

        for(let i = 0; i < nums.length; i++){
            let num = nums[i];

            if(num in map){
                return [map[num],i]
            }

            let potrebanbroj = target - nums[i]

            if(potrebanbroj in map){
                return [map[potrebanbroj],i]
            }

            map[num] = i;
        }

        return false;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        Dictionary<int, int> hashNums = new Dictionary<int, int>();

        for (int i = 0; i < numbers.Length; i++){
            if(hashNums.ContainsKey(target - numbers[i]))
                return [hashNums[target - numbers[i]]+1, i+1];
            
            hashNums.Add(numbers[i], i);
        }

        return [0,0];
    }
}
